server/src/main/java/org/opensearch/wlm/WorkloadManagementSettings.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Main class to declare Workload Management related settings
+ */
+public class WorkloadManagementSettings {
+    private static final Double DEFAULT_NODE_LEVEL_MEMORY_REJECTION_THRESHOLD = 0.8;
+    private static final Double DEFAULT_NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD = 0.9;
+    private static final Double DEFAULT_NODE_LEVEL_CPU_REJECTION_THRESHOLD = 0.8;
+    private static final Double DEFAULT_NODE_LEVEL_CPU_CANCELLATION_THRESHOLD = 0.9;
+    public static final double NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD_MAX_VALUE = 0.95;
+    public static final double NODE_LEVEL_MEMORY_REJECTION_THRESHOLD_MAX_VALUE = 0.9;
+    public static final double NODE_LEVEL_CPU_CANCELLATION_THRESHOLD_MAX_VALUE = 0.95;
+    public static final double NODE_LEVEL_CPU_REJECTION_THRESHOLD_MAX_VALUE = 0.9;
+
+    private Double nodeLevelMemoryCancellationThreshold;
+    private Double nodeLevelMemoryRejectionThreshold;
+    private Double nodeLevelCpuCancellationThreshold;
+    private Double nodeLevelCpuRejectionThreshold;
+
+    /**
+     * Setting name for node level memory based rejection threshold for QueryGroup service
+     */
+    public static final String NODE_MEMORY_REJECTION_THRESHOLD_SETTING_NAME = "query_group.node.memory_rejection_threshold";
```

Comment:
nit: should we prefix these setting names by wlm?

Reply:
we can, just wondering if that'll be too long after the prefix

Reply:
don't think so, more qualified settings are always better IMO